### PR TITLE
Add training for VideoComposer

### PR DESCRIPTION
**Pull request by SamitHuang**
Tested on 910A + MS2.0
Runnable with default setting: max_frames=8, optim=momentum.
Adapt for 910B

### CODE REVIEW

File: examples/videocomposer/vc/data/datasets.py [hunk 1]
```diff
@@ -128,3 +132,20 @@ def _get_video_train_data(self, video_key, feature_framerate, viz_mv):
         ref_frame = vit_image
 
         return ref_frame, vit_image, video_data, misc_data, mv_data
+
+
+def get_video_paths_captions(data_dir):
+    anno_list = sorted(
+        [os.path.join(data_dir, f) for f in list(filter(lambda x: x.endswith(".csv"), os.listdir(data_dir)))]
+    )
+    db_list = [pd.read_csv(f) for f in anno_list]
```
Comment (HaoyangLee) [inline examples/videocomposer/vc/data/datasets.py:141]:
```suggestion
    db_list = [pd.read_csv(f,  quotechar="\"") for f in anno_list]
```

In some dataset, the caption may consist of comma(`,`) which causes bug when loading the csv file. The column delimiter cannot be recognized correctly.
<details close markdown>

```
Traceback (most recent call last):
  File "/data4/code/mindone/examples/videocomposer/train.py", line 328, in <module>
    main(cfg)
  File "/data4/code/mindone/examples/videocomposer/train.py", line 246, in main
    dataloader = build_dataset(cfg, device_num, rank_id, tokenizer)
  File "/data4/code/mindone/examples/videocomposer/vc/data/dataset_train.py", line 172, in build_dataset
    dataset = VideoDatasetForTrain(
  File "/data4/code/mindone/examples/videocomposer/vc/data/dataset_train.py", line 64, in __init__
    video_paths, captions = get_video_paths_captions(root_dir)
  File "/data4/code/mindone/examples/videocomposer/vc/data/dataset_train.py", line 157, in get_video_paths_captions
    db_list = [pd.read_csv(f) for f in anno_list]
  File "/data4/code/mindone/examples/videocomposer/vc/data/dataset_train.py", line 157, in <listcomp>
    db_list = [pd.read_csv(f) for f in anno_list]
  File "/home/miniconda3/envs/ms21_py39/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 948, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/home/miniconda3/envs/ms21_py39/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 617, in _read
    return parser.read(nrows)
  File "/home/miniconda3/envs/ms21_py39/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 1748, in read
    ) = self._engine.read(  # type: ignore[attr-defined]
  File "/home/miniconda3/envs/ms21_py39/lib/python3.9/site-packages/pandas/io/parsers/c_parser_wrapper.py", line 234, in read
    chunks = self._reader.read_low_memory(nrows)
  File "parsers.pyx", line 843, in pandas._libs.parsers.TextReader.read_low_memory
  File "parsers.pyx", line 904, in pandas._libs.parsers.TextReader._read_rows
  File "parsers.pyx", line 879, in pandas._libs.parsers.TextReader._tokenize_rows
  File "parsers.pyx", line 890, in pandas._libs.parsers.TextReader._check_tokenize_status
  File "parsers.pyx", line 2058, in pandas._libs.parsers.raise_parser_error
pandas.errors.ParserError: Error tokenizing data. C error: Expected 2 fields in line 17, saw 3
```

</details>

A safe solution is to add **double quotation marks ("")** to the caption text in `video_caption.csv` as below.

```
video,caption
10100_ZGF3orq9BLQ.mp4,"soccer fan with hat and microphone in front of a wall"
6731_f3wUrPfbChE.mp4,"an apartment for sale in samutprakarn, phuket"
22677_HkRcWqWef9k.mp4,"four atoms with the numbers 1,2, 3, and 4,5"
...
```

File: examples/videocomposer/vc/data/dataset_train.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            mv_data[: len(frames), ...] = mvs
+
+        return vit_image, video_data, misc_data, mv_data
+
+
+def get_video_paths_captions(data_dir):
+    anno_list = sorted(
+        [os.path.join(data_dir, f) for f in list(filter(lambda x: x.endswith(".csv"), os.listdir(data_dir)))]
+    )
+    db_list = [pd.read_csv(f) for f in anno_list]
```
Comment (HaoyangLee) [inline examples/videocomposer/vc/data/dataset_train.py:157]: Refer to the comment above.

File: examples/videocomposer/demo_video/video_caption.csv [hunk 3]
```diff
@@ -0,0 +1,25 @@
+video,caption
```
Comment (HaoyangLee) [inline examples/videocomposer/demo_video/video_caption.csv:1]: Refer to the comment above.

**Review by wtomin — APPROVED** [review]
LGTM.

**Pull request merged**